interview-questions.md

```diff
@@ -8,62 +8,62 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is the difference between a parameter and an argument?
 
-  Your answer:
+  Your answer: a special variable that belongs to the function is called a parameter, and and argument is the information passed into a function where it is invoked.
 
-  Researched answer:
+  Researched answer: When you create a function, parameters are variables listed as a part of the function. and arguments are information or values passed into the function when it is invoked.
 
 
 
 2. The JavaScript built in method .map() takes predefined parameters. What are they? Which are required and which are optional?
 
-  Your answer:
+  Your answer: iterates over each element in an array and returns a new array that has the same length. a short cut version of a for loop.
 
-  Researched answer:
+  Researched answer: .map has three built in parameters, it requires one argument value. 1st one is the value of the array at the current iteration, 2. the index of the array at the current iteration and 3. the array the high order function was called on, 2nd and 3rd are optional.
 
 
 
 3. What is the difference between map and filter?
 
-  Your answer:
+  Your answer:iterates over each element in an array and returns a new array that has the same length. a short cut version of a for loop and .filter will remove what we don't want and only keep what we do wan't also returns a subset of the array and has a built in if statement
 
-  Researched answer:
+  Researched answer: .map will do something to each element in the array of the same length and .filter will make decisions about each item in an array and return a subset based on the conditions given to it
 
 
 
 4. What is the difference between a function and a method?
 
-  Your answer:
-
-  Researched answer:
+  Your answer: functions indicate action and the return is your output, functions are reusable. no other data affects the function called due to encapsulation which is information inside the curly brackets. method is a function(behavior) that belongs to an object or built in methods.
 
+  Researched answer: Functions can be customized, the input is a special variable that belongs to the function which is called a parameter. you can add logic and additional parameters seperated by commas. a function within an object is reffered to as a method which now that I think of it seem to be the exact samething but under different scenarios.
 
 
 5. What is object destructuring?
 
-  Your answer:
-
-  Researched answer:
-
+  Your answer: destructuring is defining a pathway through a nested object.
```

Comment:
It seems this could be something you could expand on.  In an interview, you want to show interest by giving detailed answers and while this is accurate, an interviewer will also be looking at how you communicate your knowledge.